### PR TITLE
feat: lint fixes for schema

### LINKED ISSUE #1322: Validate schema using linters
Tailcall should ensure consistency in building the final GraphQL schema. This is the standard it should follow:

**Case styles**
- Field names should use `camelCase`. 
- Type names should use `PascalCase`.
- Enum names should use `PascalCase`.
- Enum values should use `ALL_CAPS`, because they are similar to constants.

There are two approaches to solve this problem:
1. When the setting is enabled, we can throw a validation error that the developer see and fix using the `@modify` operator.
2. We can automatically fix it for the user.

**Proposal**
```graphql
schema @server(lint: {field: true, type: true, enum: true, enumValue: true, autoFix: false}) {
  query: Query
}
```

**Technical Requirements**
Use the `ConfigOptimizer` API defined over here https://github.com/tailcallhq/tailcall/issues/1653

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
Fixes #1322 
/claim 1322

## Summary by CodeRabbit

- **New Features**
	- Enhanced schema linting capabilities with the introduction of a new `Lint` input type and object for GraphQL configurations.
	- Added new fields and types to GraphQL schema definitions to improve testing and functionality.
	- Implemented `lint_fix` functionality across configuration modules to automate naming convention fixes.

- **Bug Fixes**
	- Adjusted numbering format in the bug report template for clearer reproduction steps.
	- Ensured consistent response header value formatting in GraphQL schema declarations.

- **Chores**
	- Removed unnecessary code block from contributing guidelines.
	- Included a new `lint` module in the public exports for configuration management.

- **Documentation**
	- Improved formatting in various test files to enhance readability and consistency.

### CODE REVIEW

File: generated/.tailcallrc.graphql [hunk 1]
```diff
@@ -633,6 +637,13 @@ enum LinkType {
   Cert
   Key
 }
+input Lint {
+  autoFix: Boolean!
+  enum: Boolean!
+  enumValue: Boolean!
+  field: Boolean!
+  type: Boolean!
+}
```
Comment (tusharmath) [inline generated/.tailcallrc.graphql:646]: all fields should be optional

File: src/config/lint.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        for field in &ty.added_fields {
+            new_ty.added_fields.push(AddField {
+                name: field.name.to_case(convert_case::Case::Camel),
+                path: field.path.clone(),
+            })
+        }
+    }
+
+    new_ty
+}
```
[the remark this answers is not in the document]
Comment (tusharmath) [reply src/config/lint.rs:74]: We can't simply rename the fields. Auto fix should be the moral equivalent of inserting modify.

[the remark this answers is not in the document]
Comment (tusharmath) [reply src/config/lint.rs:74]: Needs tests.

File: src/config/lint.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                match lint.type_lint {
+                    true => k.to_case(convert_case::Case::Pascal),
+                    false => k.clone(),
+                },
+                fix_type(lint, ty),
+            );
+        }
+        config.types = fixed_types;
+    }
+    config
```
[the remark this answers is not in the document]
Comment (webbdays) [reply src/config/lint.rs]: try.
@coderabbitai

[the remark this answers is not in the document]
Comment (webbdays) [reply src/config/lint.rs]: is tests inline with codebase?
@coderabbitai

**Comment by webbdays** [comment]
.

**Comment by ssddOnTop** [comment]
> @ssddOnTop Shall we close this PR? Nothing has happened on this PR.

Yes I wont be able to work on this till next week

**Pull request closed without merging**